gssapi/raw/ext_channel_bindings.pyx

```diff
@@ -0,0 +1,73 @@
+from gssapi.raw.cython_types cimport *
+from gssapi.raw.oids cimport *
+GSSAPI="BASE"  # This ensures that a full module is generated by Cython
+
+from gssapi.raw.misc import GSSError
+
+from gssapi.raw.sec_contexts cimport SecurityContext
+
+cdef extern from "python_gssapi_ext.h":
+    OM_uint32 gss_create_sec_context(OM_uint32 *minor_status,
+                                     gss_ctx_id_t *context)
+
+    OM_uint32 gss_set_context_flags(OM_uint32 *minor_status,
+                                    gss_ctx_id_t context,
+                                    uint64_t req_flags,
+                                    uint64_t ret_flags_understood)
+
+
+def create_sec_context():
+    """
+    create_sec_context()
+    Creates an empty SecurityContext() for use with set_context_flags(),
+    init_sec_context() and accept_sec_context().
+
+    Returns:
+        SecurityContext: an empty context
+
+    Raises:
+        GSSError
+    """
+    cdef OM_uint32 maj_stat, min_stat
+    cdef SecurityContext out_ctx = SecurityContext()
+    cdef gss_ctx_id_t ctx = GSS_C_NO_CONTEXT
+
+    maj_stat = gss_create_sec_context(&min_stat, &ctx)
+
+    if maj_stat == GSS_S_COMPLETE:
+        out_ctx.raw_ctx = ctx
+        return out_ctx
+    else:
+        raise GSSError(maj_stat, min_stat)
+
+
+def set_context_flags(SecurityContext context, uint64_t req_flags,
+                      uint64_t ret_flags_understood):
+    """
+    set_context_flags(context, req_flags, ret_flags_understood)
+    Specifies request flags and understood return flags for use with
+    init_sec_context() and accept_sec_context(). Requires a context
+    previously created by create_sec_context() and not yet used with
+    init_sec_context() or accept_sec_context().
+
+    Args:
+        context (SecurityContext): Context to set flags on
+        req_flags (uint64_t): Request flags for init/accept_sec_context()
+        ret_flags_understood: Return flags that are understood by the
```

Comment:
`ret_flags_understood` is a set of return flags that are understood by the calling application.

In other word, it is a subset of possible `ret_flag` values from `gss_init/accept_sec_context`; the only useful `ret_flag_understood` parameter to set would be the channel bound flag; setting or not setting any other flag does not change the behavior. (This last point about not changing behavior is the subject of a revision in the draft, as if you set `req_flags = GSS_C_MUTUAL_FLAG` but set `ret_flags_understood = 0`, `gss_init/accept_sec_context` will still return GSS_C_MUTUAL_FLAG on success). 

`ret_flags` was what the original [draft](https://tools.ietf.org/html/draft-ietf-kitten-channel-bound-flag-01#section-2.2.1) called it in the C bindings, hence the discrepancy in the high level API. Suggestions welcome about a more helpful name.